core/templates/domain/question/QuestionObjectFactory.ts

```diff
@@ -31,164 +35,182 @@ export interface QuestionBackendDict {
   'inapplicable_skill_misconception_ids': string[];
 }
 
-angular.module('oppia').factory('QuestionObjectFactory', [
-  'StateObjectFactory', 'DEFAULT_LANGUAGE_CODE', 'INTERACTION_SPECS',
-  function(StateObjectFactory, DEFAULT_LANGUAGE_CODE, INTERACTION_SPECS) {
-    var Question = function(
-        id, stateData, languageCode, version, linkedSkillIds,
-        inapplicableSkillMisconceptionIds) {
-      this._id = id;
-      this._stateData = stateData;
-      this._languageCode = languageCode;
-      this._version = version;
-      this._linkedSkillIds = linkedSkillIds;
-      this._inapplicableSkillMisconceptionIds = (
-        inapplicableSkillMisconceptionIds);
-    };
-
-    // ---- Instance methods ----
+export class Question {
+  _id: string;
+  _stateData: State;
+  _stateDataSchemaVersion: number;
+  _languageCode: string;
+  _version: number;
+  _linkedSkillIds: string[];
+  _inapplicableSkillMisconceptionIds: string[];
+
+  constructor(
+      id: string, stateData: State, stateDataSchemaVersion: number,
+      languageCode: string, version: number, linkedSkillIds: string[],
+      inapplicableSkillMisconceptionIds: string[]) {
+    this._id = id;
+    this._stateData = stateData;
+    this._stateDataSchemaVersion = stateDataSchemaVersion;
+    this._languageCode = languageCode;
+    this._version = version;
+    this._linkedSkillIds = linkedSkillIds;
+    this._inapplicableSkillMisconceptionIds = (
+      inapplicableSkillMisconceptionIds);
+  }
```

Comment:
I think initializing this way would be better.
```suggestion
  constructor(
      private _id: string, 
      private _stateData: State, 
      private _stateDataSchemaVersion: number,
      private _languageCode: string, 
      private _version: number,
      private _linkedSkillIds: string[],
      private _inapplicableSkillMisconceptionIds: string[]) {}
```

Reply:
Done!